swatch-core/src/main/java/org/candlepin/subscriptions/db/TallySnapshotRepository.java

```diff
@@ -77,4 +78,27 @@ Stream<TallySnapshot> findByAccountNumberAndProductIdInAndGranularityAndSnapshot
       OffsetDateTime ending);
 
   void deleteByAccountNumber(String accountNumber);
+
+  @Query(
+      "select coalesce(sum(VALUE(m)), 0.0) from TallySnapshot s "
+          + "left join s.tallyMeasurements m on key(m) = :measurementKey "
+          + "where s.accountNumber = :accountNumber and "
+          + "s.productId = :productId and "
+          + "s.granularity = :granularity and "
```

Comment:
Wouldn't this always be hourly? i.e. no need to specify as a param on the method? Not that I mind specifying it... just curious?